src/codons.py

```diff
@@ -61,7 +69,16 @@ def translate_codons(codons: list[str]) -> list[str]:
 
     """
     # FIXME: Implement the function
-    return []
+    amino_acids = []
+
+    for codon in codons:
+        if (codon.upper() in CODON_MAP):
+            aa = CODON_MAP[codon.upper()] # handles upper case 
+            amino_acids += aa
+        else:
+            return None
```

Comment:
Looks good. If you translated the codon at the first line inside the loop

```python
for codon in codons:
    codon = codon.upper()
   ...
```

you could avoid doing it twice.